app/models/user.rb

```diff
@@ -9,12 +9,23 @@ class User < ApplicationRecord
   validates :first_name, presence: { message: ErrorDescriptions::Error_List[:VALUE_CANT_BE_EMPLTY] }
   validates :last_name, presence: { message: ErrorDescriptions::Error_List[:VALUE_CANT_BE_EMPLTY] }
 
+  has_one :cart, dependent: :destroy
+
+  after_create_commit :create_new_cart
```

Comment:
Could we put this in `after_create`?

Reply:
Yes, we can. I try to put this in `after_create` and it works. That's magic. I think `after_create` just creates an object not save it in the db yet, so we can't create other relations records. We need to create a user, save it in the db, commit ok, and after that, we can create a record for the cart.

Reply:
We may want them to be created within the same transaction, rather than in separate transaction like using `after_create_commit`